### DISCUSSION TITLE
LicenseKeys section

### DISCUSSION THREAD

**Discussion by zergud** [globallightingdata/gldf#5] [General]
What is the purpose of the "LicenseKeys" section

required attribute "application" has two values - dialux and relux

what about "Open and freely available for everyone."?

**Comment by Kurpanik** [comment]
DIALux and Relux are free for everyone and the gldf format and import in both applications will be free as well. `LicenseKeys` is not intended to provide any license information for the format or tools itself, but for DIAL/RELUX partners - _should_ there be any need in the future to do so (which currently is not).

Because of this, the `LicenseKeys` element is optional. Only the attribute `application` is required, because it only makes sense to provide a key, if the application requiring it is named as well.

For further reading here are some information about partnership and benefits: [become a partner](https://www.dialux.com/en-GB/become-a-partner ) (likewise for RELUX on their website). And a list of our partner [here](https://www.dialux.com/en-GB/dialux-partner )

**Reply by zergud** [reply]
to be clear:
are there any license restrictions for CREATE and USE gldf and l3d formats by "third party" application?
(as the GLDF intended to be global interchanging format for lighting fixtures) 

some phrases in xsd and documentation are confusing 
for example "Checksum for structure and format. Could be signed with a tool developed by Relux and Dial" 
- checksum algorithm? 
- can "could" turn into "should" in future?

most likely such questions will disappear as the format specification is approved and gltf samples provided :)

**Reply by Kurpanik** [reply]
No, there aren't any license restrictions for CREATE and USE gldf and l3d formats by "third party" application. Gldf/L3d is relaesed under MIT license and you are free to do with it, whatever you want.

The checksum is also not intended for licensing, but to protect manufacturer product data from being manipulated. An application can evaluate it and check whether a manufacturer's product data or lighting information has been changed and indicate this. DIALux and RELUX will probably implement this. But again: this will be
1) always optional
2) verfication will be possible for any 3rd party application as well, as we will disclose the algorithm (standard HMAC/SHA256) and manufacturers can of course publish their public keys or we will do on gldf.io

However, it is possible, that the provision of an infrastructure for the signing itself will be not offered free of charge. But as I've written above: it **will be always optional and is not intended for any licensing**, but rather for manufacturers who wants to protect their product data from manipulation.

**Reply by zergud** [reply]
thank you

**Reply by Kurpanik** [reply]
You're welcome.
